chef/automate
# adding proxy host logic

Signed-off-by: Daniel Bright <danielcbright@gmail.com>

### :nut_and_bolt: Description: What code changed, and why?

I updated the `automate-es-gateway` logic to allow for the setting of the `proxy_set_header Host` value to be one of the users choosing. This will allow for external ES to work when using an external cluster that uses hostname based routing (think most Nginx load balancers, Elastic Cloud, and probably more). It sets the default to `$http_host` if it is not defined, which is the current behaviour.

To enable the setting you would add this to a `config.toml` during deploy or via a config patch:

```
[esgateway.v1.sys.ngx.http]
proxy_set_header_host = "yourawesomeexternaleshostname:1234"
```

### :chains: Related Resources

The `esgateway` api config was also updated to reflect the changes.

### :+1: Definition of Done

You can make the above changes, and they are reflected in the `automate-es-gateway`'s final, rendered `nginx.conf` in the `/hab/svc/automate-es-gateway/config/nginx.conf` file.

### :athletic_shoe: How to Build and Test the Change

1. `start_all_services`
2. create a custom `patch.toml` with the above settings
3. patch the chef-automate install
4. verify that the `/hab/svc/automate-es-gateway/config/nginx.conf` reflects the changes

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
